homeassistant/loader.py

```diff
@@ -1077,43 +1124,35 @@ def _get_platform_cached(self, full_name: str) -> ModuleType | None:
 
     def get_platform(self, platform_name: str) -> ModuleType:
         """Return a platform for an integration."""
-        if platform := self._get_platform_cached(f"{self.domain}.{platform_name}"):
+        if platform := self._get_platform_cached_or_raise(
+            f"{self.domain}.{platform_name}"
+        ):
             return platform
         return self._load_platform(platform_name)
```

Comment:
Why not check platform_exists ?

Reply:
It needs to raise to preserve api compatibility.

Reply:
But can't we raise without going through importlib ?

Reply:
Discussed on discord about the missing platform already being cached